app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,74 @@
+module Products
+
+  class UserNoteMode
+
+    VALID_MODES = %w(hidden optional required).freeze
+
+    attr_reader :raw_value
+
+    # An array of UserNoteModes representing the possible modes
+    # Use this in validations and form collections.
+    def self.all
+      VALID_MODES.map { |value| new(value) }
+    end
+
+    def self.from_string(mode_string)
```

Comment:
I don't mind `from_string`, although the `[]` option would be okay too. Either way we go, though, I'd like it if we added a comment that this is the preferred option for the public API over `UserNoteMode.new`.

I do think this is a nice solution to the hard/soft error problem.